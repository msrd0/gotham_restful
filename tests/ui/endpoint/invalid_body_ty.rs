#[macro_use]
extern crate gotham_restful;
use gotham_restful::gotham::hyper::Method;

#[derive(Resource)]
#[resource(endpoint)]
struct FooResource;

#[derive(Debug)]
struct FooBody {
	foo: String
}

#[endpoint(method = "Method::GET", uri = "", body = true)]
fn endpoint(_: FooBody) {
	unimplemented!()
}

fn main() {}
