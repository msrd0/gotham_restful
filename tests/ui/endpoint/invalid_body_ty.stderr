error[E0277]: the trait bound `FooBody: OpenapiType` is not satisfied
  --> tests/ui/endpoint/invalid_body_ty.rs:15:16
   |
15 | fn endpoint(_: FooBody) {
   |                ^^^^^^^ the trait `OpenapiType` is not implemented for `FooBody`
   |
  ::: src/endpoint.rs:80:16
   |
80 |     type Body: RequestBody + Send;
   |                ----------- required by this bound in `gotham_restful::EndpointWithSchema::Body`

error[E0277]: the trait bound `for<'de> FooBody: serde::de::Deserialize<'de>` is not satisfied
  --> tests/ui/endpoint/invalid_body_ty.rs:15:16
   |
15 | fn endpoint(_: FooBody) {
   |                ^^^^^^^ the trait `for<'de> serde::de::Deserialize<'de>` is not implemented for `FooBody`
   |
  ::: src/endpoint.rs:80:16
   |
80 |     type Body: RequestBody + Send;
   |                ----------- required by this bound in `gotham_restful::EndpointWithSchema::Body`
   |
   = note: required because of the requirements on the impl of `serde::de::DeserializeOwned` for `FooBody`
   = note: required because of the requirements on the impl of `gotham_restful::RequestBody` for `FooBody`
